rest/taskana-rest-spring/src/main/java/pro/taskana/rest/WorkbasketController.java

```diff
@@ -93,18 +93,27 @@
         @RequestParam(value = "type", required = false) String type,
         @RequestParam(value = "requiredPermission", required = false) String requiredPermission,
         @RequestParam(value = "page", required = false) String page,
-        @RequestParam(value = "pagesize", required = false) String pagesize) {
+        @RequestParam(value = "pagesize", required = false) String pagesize) throws InvalidArgumentException {
 
         WorkbasketQuery query = workbasketService.createWorkbasketQuery();
         addSortingToQuery(query, sortBy, order);
         addAttributeFilter(query, name, nameLike, key, keyLike, descLike, owner, ownerLike, type);
         addAuthorizationFilter(query, requiredPermission);
 
-        // long totalElements = query.count();
-
-        PageMetadata pageMetadata = new PageMetadata(Integer.MAX_VALUE, 0, 11, 4);
-        List<WorkbasketSummary> workbasketSummaries = query.listPage((int) pageMetadata.getNumber(),
-            (int) pageMetadata.getSize());
+        PageMetadata pageMetadata = null;
```

Comment:
I would add all this pagination logic into a function. AddPagingToQuery or something like this.

Reply:
I considered this. But this part actually also contains the non-paging part. Therefore this does not work. Since I expect a larger refactoring as soon as we do have another paging resource I would think about a better design at that point in time.